File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/EmailModel.java
package expo.modules.contacts.models;

import android.database.Cursor;
import android.provider.ContactsContract;

import expo.modules.contacts.EXColumns;

public class EmailModel extends BaseModel {

  @Override
  public String getContentType() {
    return ContactsContract.CommonDataKinds.Email.CONTENT_ITEM_TYPE;
  }

  @Override
  public String getDataAlias() {
    return "email";
  }

  @Override
  protected String getLabelFromCursor(Cursor cursor) {
    String label = super.getLabelFromCursor(cursor);
    if (label != null) return label;
    switch (cursor.getInt(cursor.getColumnIndex(EXColumns.TYPE))) {
      case ContactsContract.CommonDataKinds.Email.TYPE_HOME:
        return "home";
      case ContactsContract.CommonDataKinds.Email.TYPE_WORK:
        return "work";
      case ContactsContract.CommonDataKinds.Email.TYPE_MOBILE:
        return "mobile";
      case ContactsContract.CommonDataKinds.Email.TYPE_OTHER:
        return "other";
      default:
        return "unknown";
    }
  }
}

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/EmailModel.kt
package expo.modules.contacts.models

import android.database.Cursor
import android.provider.ContactsContract.CommonDataKinds
import expo.modules.contacts.Columns

class EmailModel : BaseModel() {
  override val contentType: String = CommonDataKinds.Email.CONTENT_ITEM_TYPE
  override val dataAlias: String = "email"

  override fun mapStringToType(label: String?): Int {
    val mailType = when (label) {
      "home" -> CommonDataKinds.Phone.TYPE_HOME
      "mobile" -> CommonDataKinds.Phone.TYPE_MOBILE
      "work" -> CommonDataKinds.Phone.TYPE_WORK
      "other" -> CommonDataKinds.Phone.TYPE_OTHER
      else -> Columns.TYPE_CUSTOM
    }
    return mailType
  }

  override fun getLabelFromCursor(cursor: Cursor): String {
    val label = super.getLabelFromCursor(cursor)
    return label
      ?: when (cursor.getInt(cursor.getColumnIndexOrThrow(Columns.TYPE))) {
        CommonDataKinds.Email.TYPE_HOME -> "home"
        CommonDataKinds.Email.TYPE_WORK -> "work"
        CommonDataKinds.Email.TYPE_MOBILE -> "mobile"
        CommonDataKinds.Email.TYPE_OTHER -> "other"
        else -> "unknown"
      }
  }
}
